Fixes #11 (#14)
Kotlin/api-guidelines: docs/topics/jvm-api-guidelines-backward-compatibility.md

Sometimes, especially when you don't use [explicit API mode](whatsnew14.md#explicit-api-mode-for-library-authors), 
a return type declaration can change implicitly. But even if it's not the case, you might want to narrow the signature. 
For example, sometimes you realize that you need index access to the elements of your collection and want to change 

> the return type from `Collection` to `List`. Widening the return type will usually break the source compatibility, for example convertting from `List` to `Collection` will break all the code using index access. Narrowing return types is usually source-compatible change, but it breaks the binary compatibility and here we will describe how.

Reviewer: Please, create lines no longer than 120 symbols; it allows folks to work with text from a laptop screen conveniently :)

Reviewer: @asm0dey should I expect a commit on separating this long line into several?

Author: Sure!

Author: And fixed!

> the return type from `Collection` to `List`. Widening the return type will usually break the source compatibility, for example convertting from `List` to `Collection` will break all the code using index access. Narrowing return types is usually source-compatible change, but it breaks the binary compatibility and here we will describe how.

Reviewer: ```suggestion
the return type from `Collection` to `List`. Widening the return type usually breaks the source compatibility; for example, converting from `List` to `Collection` breaks all the code using index access. Narrowing return types is usually a source-compatible change, but it breaks the binary compatibility, and this section describes how.
```
